mmv1/third_party/terraform/website/docs/d/compute_region_instance_template.html.markdown

```diff
@@ -0,0 +1,305 @@
+    instance.
+
+* `preemptible` - Allows instance to be preempted. This defaults to
+    false. Read more on this
+    [here](https://cloud.google.com/compute/docs/instances/preemptible).
+
+* `node_affinities` - Specifies node affinities or anti-affinities
+   to determine which sole-tenant nodes your instances and managed instance
+   groups will use as host systems. Read more on sole-tenant node creation
+   [here](https://cloud.google.com/compute/docs/nodes/create-nodes).
+   Structure [documented below](#nested_node_affinities).
+   
+* `provisioning_model` - Describe the type of preemptible VM. 
+
+* `instance_termination_action` - Describe the type of termination action for `SPOT` VM. Can be `STOP` or `DELETE`.  Read more on [here](https://cloud.google.com/compute/docs/instances/create-use-spot) 
+
+<a name="nested_guest_accelerator"></a>The `guest_accelerator` block supports:
+
+* `type` - The accelerator type resource to expose to this instance. E.g. `nvidia-tesla-k80`.
+
+* `count` - The number of the guest accelerator cards exposed to this instance.
+
+<a name="nested_node_affinities"></a>The `node_affinities` block supports:
+
+* `key` - The key for the node affinity label.
+
+* `operator` - The operator. Can be `IN` for node-affinities
+    or `NOT_IN` for anti-affinities.
+
+* `value` - The values for the node affinity label.
+
+<a name="nested_shielded_instance_config"></a>The `shielded_instance_config` block supports:
+
+* `enable_secure_boot` -- Verify the digital signature of all boot components, and halt the boot process if signature verification fails. Defaults to false.
+
+* `enable_vtpm` -- Use a virtualized trusted platform module, which is a specialized computer chip you can use to encrypt objects like keys and certificates. Defaults to true.
+
+* `enable_integrity_monitoring` -- Compare the most recent boot measurements to the integrity policy baseline and return a pair of pass/fail results depending on whether they match or not. Defaults to true.
+
+<a name="nested_confidential_instance_config"></a>The `confidential_instance_config` block supports:
+
+* `enable_confidential_compute` Defines whether the instance should have confidential compute enabled. [`on_host_maintenance`](#on_host_maintenance) has to be set to TERMINATE or this will fail to create the VM.
+
+<a name="nested_network_performance_config"></a>The `network_performance_config` block supports:
+
+* `total_egress_bandwidth_tier` - The egress bandwidth tier for the instance.
+
+---
+
+* `id` - an identifier for the resource with format `projects/{{project}}/regions/{{region}}/instanceTemplates/{{name}}`
+
+* `metadata_fingerprint` - The unique fingerprint of the metadata.
+
+* `self_link` - The URI of the created resource.
+
+* `tags_fingerprint` - The unique fingerprint of the tags.
+
+[1]: /docs/providers/google/r/compute_instance_group_manager.html
+[2]: /docs/configuration/resources.html#lifecycle
```

Comment:
What is the reason to add the docs here? I think we can delete them.

Reply:
It's the user guide for the new data source. Do you mean it's the wrong place to place this doc?

Reply:
The two docs are not needed, as the user does not need to know the code source.

Reply:
Oh, I see. Corrected. Thanks for catching it!